### PR TITLE
test: test run locally

### PR DESCRIPTION
**Pull request by IlyasYOY**
Hello! Thank you for this plugin! It's awesome! 

I saw your comment [here](https://github.com/nvim-lua/plenary.nvim/issues/474#issuecomment-1579104083). So, I decided to give it a try.

I managed to run tests locally. They didn't hang. 

But some of them were failing: 

```
~/Projects/other/nvim-surrond-fork on  main ⌚ 21:47:42
$ make | grep -e "Success:" -e "Failed :"
Starting...Success:     0
Failed :        2
Success:        0
Failed :        3
Success:        0
Failed :        5
Success:        0
Failed :        8
Success:        1
Failed :        6
Success:        3
Failed :        16
Success:        9
Failed :        0
```

PS. I'm sure that this is not the solution, but hope this may help you find the solution.

### CODE REVIEW

**Comment by IlyasYOY** [comment]
In my plugins I look up to @folke's solutions. Good example is shown [here](https://github.com/folke/noice.nvim).

**Comment by kylechui** [comment]
I checked out your branch and it also doesn't hang anymore, although now it is failing a bunch of test cases, which is less than ideal (and also strange, given that the core plugin doesn't utilize plenary at all). My tests work if I run them file-by-file, and they used to work on an older version of Neovim; I really don't know what happened to it. I do appreciate your work; hopefully I can understand your commit better at some point and isolate what the issue is.

**Comment by IlyasYOY** [comment]
I did fix the issue. The answer is always simple.

I forgot to add: `require("nvim-surround").setup()`.

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -35,10 +35,6 @@ jobs:
           neovim: true
           version: nightly
 
-      - name: Install plenary
```
Comment (IlyasYOY) [inline .github/workflows/integration.yml:38]: now we load plenary from lua: https://github.com/kylechui/nvim-surround/pull/246/files#diff-dff985f3ac32aac36a5ec14df2bba7e6a8c063293e4ea1a83cb1587d0a3eebbaR11

File: .github/workflows/integration.yml [hunk 2]
```diff
@@ -35,10 +35,6 @@ jobs:
...
-        run: |
-          git clone --depth 1 https://github.com/nvim-lua/plenary.nvim ~/.local/share/nvim/site/pack/vendor/start/plenary.nvim
-          ln -s $(pwd) ~/.local/share/nvim/site/pack/vendor/start
-
       - name: Run tests
-        run: nvim --headless --noplugin -c 'packadd plenary.nvim' -c "PlenaryBustedDirectory tests/ { minimal_init = 'tests/minimal_init.lua' }"
+        run: make test_lua
```
Comment (IlyasYOY) [inline .github/workflows/integration.yml:39]: reuse logic from `Makefile` so it's not duplicated.

PS. Maybe you think I should remove `Makefile`

**Comment by kylechui** [comment]
I'm not entirely sure what happened, but I updated plenary and it seems to be working for me again(?) Thanks for taking the time to submit the PR nonetheless (and sorry for the radio silence)!

**Pull request closed without merging**